app/src/main/java/org/mifos/mobilebanking/ui/activities/SettingsActivity.java

```diff
@@ -16,12 +18,24 @@ protected void onCreate(Bundle savedInstanceState) {
         setToolbarTitle(getString(R.string.settings));
         showBackButton();
         replaceFragment(SettingsFragment.newInstance(), false, R.id.container);
+        Intent intent = getIntent();
+        if (intent.hasExtra("hasSettingsChanged")) {
```

Comment:
Don't use hard-coded strings. Instead, define constants in `Constants.java` and Don't create an object for Intent. Just use `getIntent()` and make the method calls.